ninja-rocker/src/main/java/ninja/rocker/TemplateEngineRocker.java

```diff
@@ -0,0 +1,235 @@
+        this.commonErrorTemplates.put(NinjaConstant.LOCATION_VIEW_FTL_HTML_BAD_REQUEST, common_error.class);
+        this.commonErrorTemplates.put(NinjaConstant.LOCATION_VIEW_FTL_HTML_FORBIDDEN, common_error.class);
+        this.commonErrorTemplates.put(NinjaConstant.LOCATION_VIEW_FTL_HTML_NOT_FOUND, common_error.class);
+        this.commonErrorTemplates.put(NinjaConstant.LOCATION_VIEW_FTL_HTML_UNAUTHORIZED, common_error.class);
+ 
+        loadCustomErrorTemplates();
+    }
+    
+    public final void loadCustomErrorTemplates() {
+        Class<NinjaRockerTemplate> internalServerErrorClass = getCustomErrorViewClass("views.system.internal_server_error");
+        if (internalServerErrorClass != null) {
+            this.commonErrorTemplates.put(NinjaConstant.LOCATION_VIEW_FTL_HTML_INTERNAL_SERVER_ERROR, internalServerErrorClass);
+        }
+        
+        Class<NinjaRockerTemplate> badRequestClass = getCustomErrorViewClass("views.system.bad_request");
+        if (badRequestClass != null) {
+            this.commonErrorTemplates.put(NinjaConstant.LOCATION_VIEW_FTL_HTML_BAD_REQUEST, badRequestClass);
+        }
+        
+        Class<NinjaRockerTemplate> forbiddenClass = getCustomErrorViewClass("views.system.forbidden");
+        if (forbiddenClass != null) {
+            this.commonErrorTemplates.put(NinjaConstant.LOCATION_VIEW_FTL_HTML_FORBIDDEN, forbiddenClass);
+        }
+        
+        Class<NinjaRockerTemplate> notFoundClass = getCustomErrorViewClass("views.system.not_found");
+        if (notFoundClass != null) {
+            this.commonErrorTemplates.put(NinjaConstant.LOCATION_VIEW_FTL_HTML_NOT_FOUND, notFoundClass);
+        }
+        
+        Class<NinjaRockerTemplate> unauthorizedClass = getCustomErrorViewClass("views.system.unauthorized");
+        if (unauthorizedClass != null) {
+            this.commonErrorTemplates.put(NinjaConstant.LOCATION_VIEW_FTL_HTML_UNAUTHORIZED, unauthorizedClass);
+        }
+    }
+    
+    public Class<NinjaRockerTemplate> getCustomErrorViewClass(String className) {
+        try {
+            return (Class<NinjaRockerTemplate>)Class.forName(className);
+        } catch (Throwable t) {
+            // do nothing
+            return null;
+        }
+    }
+    
+    @Override
+    public String getContentType() {
+        return this.contentType;
+    }
+
+    @Override
+    public String getSuffixOfTemplatingEngine() {
+        return this.fileSuffix;
+    }
+
+    @Override
+    public void invoke(Context context, Result result) {
+        
+        Object object = result.getRenderable();
+        
+        // various types of objects renderable may store
+        Map<String,Object> valueMap = null;
+        Message valueMessage = null;
+        NinjaRockerTemplate rockerTemplate = null;
+        
+        // if the object is null we simply render an empty map...
+        if (object == null) {            
+            valueMap = Collections.EMPTY_MAP;
+        }
```

Comment:
Ninja uses this  K&R style... so else should be always on the same line as } ... We should try to keep the codebase uniform...